File: src/java/src/main/java/org/elastos/hive/vault/database/FindOptions.java
package org.elastos.hive.vault.database;

public class FindOptions {
	private Integer skip;
	private Integer limit;

	public FindOptions setSkip(Integer skip) {
		this.skip = skip;
		return this;
	}

	public FindOptions setLimit(Integer limit) {
		this.limit = limit;
		return this;
	}

	public Integer getSkip() {
		return skip;
	}

	public String getSkipStr() {
		return skip != null ? String.valueOf(skip) : "";
	}

	public Integer getLimit() {
		return limit;
	}

	public String getLimitStr() {
		return limit != null ? String.valueOf(limit) : "";
	}
}

File: src/service/database/findoptions.ts
/**
 * The options for the finding operation.
 */
export class FindOptions {
    private skip: number;
    private limit: number;

    setSkip(skip: number) {
        this.skip = skip
        return this;
    }

    setLimit(limit: number) {
        this.limit = limit
        return this;
    }

    getSkip(): number {
        return this.skip;
    }

    getLimit(): number {
        return this.limit;
    }
}